README.md
<<<<<<< HEAD
<<<<<<< HEAD
# Startup Pro

[![WordPress](https://img.shields.io/badge/wordpress-4.9.8%20tested-brightgreen.svg)]() [![License](https://img.shields.io/badge/license-GPL--3.0--or--later-blue.svg)](https://github.com/seothemes/startup-pro/blob/master/LICENSE.md)

A developer-friendly starter theme used for creating commercial child themes for the Genesis Framework.

It uses Composer to pull in the [Core](https://github.com/seothemes/core) component library which provides the PHP logic for the theme's configuration, and it uses [Gulp WP Toolkit](https://github.com/craigsimps/gulp-wp-toolkit) to automate mundane build tasks like compiling SCSS and minifying images.

Check out the [live demo](https://demo.seothemes.com/genesis-starter)

<img src="https://seothemes.com/wp-content/uploads/2018/09/starter-screenshot.png" alt="Startup Pro screenshot" width="500">

## Table of Contents

* [Features](#features)
* [Requirements](#requirements)
* [Installation](#installation)
    * [One line command](#one-line-command)
    * [Individual commands](#individual-commands)
* [Setup](#setup)
* [Usage](#usage)
* [Development](#development)
* [Structure](#structure)
* [Contributing](#contributing)
* [Authors](#authors)
* [Special Thanks](#special-thanks)

## Features

The Startup Pro aims to modernize, organize and enhance some aspects of Genesis child theme development. Take a look at what is waiting for you:

- [Bourbon](https://github.com/seothemes/startup-pro/tree/master/assets/styles) as a lightweight Sass toolkit
- [Gulp](https://gulpjs.com/) for automating development build tasks
- [Browsersync](https://browsersync.io/) for synchronized browser testing
- [Config-based](https://www.alainschlesser.com/config-files-for-reusable-code/), OOP modular architecture
- [CLI setup script](#setup) to automatically update information
- [Yarn](https://yarnpkg.com/lang/en/docs/install/#mac-stable) or [NPM](https://www.npmjs.com/) for managing Node dependencies
- [Composer](https://getcomposer.org/) for managing PHP dependencies
- [PSR-4](https://www.php-fig.org/psr/psr-4/) class autoloading
- [Namespaced](http://php.net/manual/en/language.namespaces.basics.php) to avoid naming conflicts

## Requirements

| Requirement | How to Check | How to Install |
| :---------- | :----------- | :------------- |
| PHP >= 5.4 | `php -v` | [php.net](http://php.net/manual/en/install.php) |
| WordPress >= 4.8 | `Admin Footer` | [wordpress.org](https://codex.wordpress.org/Installing_WordPress) |
| Genesis >= 2.6 | `Theme Page` | [studiopress.com](http://www.shareasale.com/r.cfm?b=346198&u=1459023&m=28169&urllink=&afftrack=) |
| Composer >= 1.5.0 | `composer --version` | [getcomposer.org](https://getcomposer.org/doc/00-intro.md#installation-linux-unix-osx) |
| Node >= 9.10.1 | `node -v` | [nodejs.org](https://nodejs.org/) |
| NPM >= 5.6.0 | `npm -v` | [npm.js](https://www.npmjs.com/) |
| Yarn >= 0.2.x | `yarn -v` | [yarnpkg.com](https://yarnpkg.com/lang/en/docs/install/#mac-stable) |
| Gulp CLI >= 1.3.0 | `gulp -v` | [gulp.js](https://gulpjs.com/) |
| Gulp = 3.9.1 | `gulp -v` | [gulp.js](https://gulpjs.com/) |

## Installation

### One line command:

Install the latest development version of the Startup Pro using Composer from your WordPress themes directory (replace `your-theme-name` below with the name of your theme):

```shell
composer create-project seothemes/startup-pro your-theme-name dev-master && cd "$(\ls -1dt ./*/ | head -n 1)" && sh setup.sh
```

### Individual commands:

Install the latest development version of the Startup Pro using Composer from your WordPress themes directory (replace `your-theme-name` below with the name of your theme):

```shell
composer create-project seothemes/startup-pro your-theme-name dev-master
```

Navigate into the theme's root directory:

```shell
cd your-theme-name
```

Run the setup script to rename the theme, build the theme assets and kick-off BrowserSync:

```shell
sh setup.sh
```

## Setup

The Startup Pro includes a powerful setup script which automates the process of updating theme details:

<a href="https://github.com/seothemes/startup-pro/blob/master/setup.sh" target="_blank"><img src="https://seothemes.com/wp-content/uploads/2018/07/startup-pro-setup-script.png" alt="Startup Pro setup script" width="500"></a>

It replaces the following details with your own:

- Theme name
- Theme textdomain
- Theme author
- Theme author URL
- Theme development URL
- Theme namespace
- Theme version

## Usage

The Startup Pro is intended to be used with [SEO Themes Core](https://packagist.org/packages/seothemes/core). All changes to the child theme should be made via the theme configuration file. This can be used to change almost every aspect of the theme, including theme features, navigation menus, image sizes, widget areas and more. An example config file is included with this theme.

Components are only loaded when a config key is provided. They can be added or removed depending on the requirements of your project. For example, to remove the PluginActivation component, simply remove it's config key from the return statement in `config/defaults.php`.

The `app` directory is provided to house project-specific PHP files if additional functionality is required. It comes pre-configured with PSR-4 autoloading. Custom components can be created in the same way other Core components are, by first creating a class which extends the `Component` class and then adding it's configuration to the theme's config file. An [Example Component](https://github.com/seothemes/startup-pro/blob/master/app/Example.php) is included with the theme to get you started.

Project details such as theme name, author, version number etc should only ever be changed from the `package.json` file. The Gulp build task reads this file and automatically places the relevant information to the correct locations throughout the theme. 

Static assets are organized in the `resources` directory. This folder contains theme scripts, styles, images, fonts, views and language translation files.

## Development

Please refer to the [Gulp WP Toolkit Instructions](https://github.com/craigsimps/gulp-wp-toolkit#tasks) for a complete list of available build tasks.

In addition to Gulp WP Toolkit's tasks, there is also a `zip` task which can be used to generate an archive of your theme, including the required composer package files and none of the unnecessary files. The list of included files can be modified from the `toolkit.extendConfig.src.zip` config in the Gulpfile.

## Contributing

Contributions are welcome from everyone. We have [contributing guidelines](https://github.com/seothemes/startup-pro/blob/master/.github/CONTRIBUTING.md) to help you get started.

## Authors

<a href="https://seothemes.com" target="_blank"><img src="https://seothemes.com/wp-content/uploads/2018/07/seothemes-startup-pro.png" alt="SEO Themes logo" width="150"></a> &nbsp; <a href="https://github.com/d2themes" target="_blank"><img src="https://seothemes.com/wp-content/uploads/2018/08/d2themes.png" alt="D2 Themes logo" width="150"></a>

See also the list of [contributors](https://github.com/seothemes/startup-pro/graphs/contributors) who participated in this project.

## Special Thanks

A shout out to anyone who's code was used in or provided inspiration to this project:

<a href="https://github.com/craigsimps/" target="_blank">Craig Simpson</a>, 
<a href="https://github.com/christophherr/" target="_blank">Christoph Herr</a>, 
<a href="https://github.com/garyjones/" target="_blank">Gary Jones</a>, 
<a href="https://github.com/timothyjensen/" target="_blank">Tim Jensen</a>, 
<a href="https://github.com/billerickson/" target="_blank">Bill Erickson</a>, 
<a href="https://github.com/srikat/" target="_blank">Sridhar Katakam</a>, 
<a href="https://github.com/nathanrice/" target="_blank">Nathan Rice</a>, 
<a href="https://github.com/bgardner/" target="_blank">Brian Gardner</a>
=======
# Masa Theme

Github project link: https://github.com/fasterwp/masa/

## Theme Support

Please visit <https://fasterwp.com/contact/> for theme support.

## Installation Instructions

### Install Genesis Framework

### Install and activate Masa child theme (masa.zip)

1. Upload the Genesis Sample theme folder via FTP to your wp-content/themes/ directory. (The Genesis parent theme needs to be in the wp-content/themes/ directory as well.)
2. Go to your WordPress dashboard and select Appearance.
3. Activate the Masa theme.
4. Inside your WordPress dashboard, go to Genesis > Theme Settings and configure theme to your liking.

### Install and activate recommended plugins

- Genesis eNews Extended
- Genesis Simple Edits
- Ninja Forms
- AJAX Thumbnail Rebuild
- Genesis Columns Advanced
- Genesis Simple FAQ
- Genesis Testimonial Slider
- Widget Importer & Exporter
- Simple Social Icons

### Import demo content

You can find the demo content in the /xml folder (file name masa-demo-content.xml).
Inside your WordPress dashboard, go to Tools > Import, install WordPress Importer if necessary and run the importer.

If you installed and activated the Widget Importer & Exporter plugin go to Tools > Widget Importer & Exporter, select the masa-widgets.wie file from the /xml folder and Import widgets.

### Setup Genesis, widgets & all

Inside your WordPress dashboard, go to Genesis > Theme Settings and configure to your liking.
Inside your WordPress dashboard, go to Appearance > Widgets and configure to your liking.
Inside your WordPress dashboard, go to Appearance > Customizer and configure to your liking.
>>>>>>> parent of fa51bee... update readme
=======
# Simplenet Theme

forked Masa theme: https://github.com/fasterwp/masa/
>>>>>>> parent of 221c886... new base - startup pro


CHANGELOG.md
# Startup Pro Changelog

## [1.1.6] - 2018-11-24
* Fixed: Breadcrumbs on product and download shortcode pages
* Fixed: Some breadcrumbs not clickable
* Fixed: Only add animations to front-page

## [1.1.5] - 2018-11-17
* Added: Featured content styles for homepage
* Changed: Improved breadcrumb display
* Fixed: Angle divider 1px line on some devices
* Fixed: Duplicate post meta on archives
* Fixed: Sidebar overlay z-index issue
* Fixed: Curve divider in Firefox
* Fixed: Mobile menu broken on inner pages

## [1.1.4] - 2018-11-13
* Added: Front page fade-in animations
* Added: Page Builder template
* Added: Setting for default hero image
* Fixed: Removed namespace from Landing Page template

## [1.1.3] - 2018-11-10
* Added: Hero section background option
* Added: Tooltips to some customizer settings
* Added: Hero image size
* Changed: Entry-title to hero-title
* Changed: Entry-subtitle to hero-subtitle
* Changed: Refactored hero section functions
* Fixed: Atomic blocks posts width

## [1.1.2] - 2018-11-09
* Added: Front Page 1 video style setting
* Fixed: Icon Widget defaults

## [1.1.1] - 2018-11-06
* Fixed: Incorrect script dependencies

## [1.1.0] - 2018-11-06
* Changed: Complete rewrite
* Removed: Composer packages

## [1.0.0] - 2018-11-05
* Added: Initial release


includes/kirki.php
<?php
/**
 * Startup Pro
 *
 * This file contains Customizer functionality for the child theme.
 *
 * @package   SeoThemes\StartupPro
 * @link      https://seothemes.com/startup-pro
 * @author    SEO Themes
 * @copyright Copyright © 2018 SEO Themes
 * @license   GPL-3.0-or-later
 */

// If this file is called directly, abort..
if ( ! defined( 'WPINC' ) ) {
	die;
}

add_filter( 'kirki/dynamic_css/method', 'startup_kirki_output_method' );
/**
 * Set the Kirki CSS output method.
 *
 * @since 1.1.0
 *
 * @return string
 */
function startup_kirki_output_method() {
	return 'file';
}

add_filter( 'kirki_config', 'startup_kirki_remove_loader' );
/**
 * Removes Kirki loader image.
 *
 * @since  1.1.0
 *
 * @return array
 */
function startup_kirki_remove_loader() {
	return wp_parse_args( [
		'disable_loader' => true,
	] );
}

add_action( 'after_setup_theme', 'startup_kirki_config' );
/**
 * Adds the theme's Kirki config.
 *
 * @since  1.1.0
 *
 * @link   https://aristath.github.io/kirki/docs/getting-started/config.html
 *
 * @return void
 */
function startup_kirki_config() {
	if ( ! class_exists( 'Kirki' ) ) {
		return;
	}

	Kirki::add_config( CHILD_THEME_HANDLE, array() );
}

add_action( 'customize_register', 'startup_customizer_defaults', 99 );
/**
 * Modify default Customizer settings.
 *
 * @since  1.1.0
 *
 * @param  object $wp_customize Global customizer object.
 *
 * @return void
 */
function startup_customizer_defaults( $wp_customize ) {
	$wp_customize->remove_control( 'background_color' );
	$wp_customize->remove_control( 'header_textcolor' );
	$wp_customize->get_control( 'header_text' )->label = __( 'Display Site Title', 'startup-pro' );
}

add_action( 'after_setup_theme', 'startup_kirki_register' );
/**
 * Register panels, sections and fields.
 *
 * @since  1.1.0
 *
 * @return void
 */
function startup_kirki_register() {
	if ( ! class_exists( 'Kirki' ) ) {
		return;
	}

	Kirki::add_section( CHILD_THEME_HANDLE, [
		'title'    => __( 'Startup Pro Settings', 'startup-pro' ),
		'priority' => 150,
	]);

	$fields = require_once CHILD_THEME_DIR . '/config/kirki.php';

	foreach ( $fields as $field => $args ) {
		Kirki::add_field( CHILD_THEME_HANDLE, $args );
	}
}

add_action( 'customize_controls_print_styles', 'startup_kirki_admin_styles', 99 );
/**
 * Adds custom styles to the WordPress Customizer.
 *
 * @since  1.1.0
 *
 * @return void
 */
function startup_kirki_admin_styles() {
	?>
    <style id="startup-pro-inline-css">
        .wp-color-result .color-alpha {
            height: 22px !important;
        }

        .wp-picker-active .iris-picker {
            margin-bottom: 12px;
        }

        .customize-control-kirki-dimensions .wrapper .control > div h5 {
            margin-top: 0;
        }

        .customize-control-kirki-multicolor .wp-picker-container > .wp-color-result {
            width: auto;
            min-width: 110px;
        }

        .wp-color-result-text {
            text-align: left !important;
        }

        .tooltip-wrapper .tooltip-content {
            background: #555d66 !important;
            color: #fff !important;
        }

        .tooltip-wrapper .tooltip-content:after {
            color: #555d66 !important;
        }
    </style>
	<?php
}

add_action( 'customize_register', 'startup_kirki_installer_register', 999 );
/**
 * Registers the section, setting & control for the kirki installer.
 *
 * @since 1.1.0
 *
 * @param object $wp_customize The main customizer object.
 *
 * @return void
 */
function startup_kirki_installer_register( $wp_customize ) {

	// Early exit if Kirki exists.
	if ( class_exists( 'Kirki' ) ) {
		return;
	}

	if ( class_exists( 'WP_Customize_Section' ) && ! class_exists( 'Kirki_Installer_Section' ) ) {
		/**
		 * Recommend the installation of Kirki using a custom section.
		 *
		 * @see WP_Customize_Section
		 */
		class Kirki_Installer_Section extends WP_Customize_Section {

			/**
			 * Customize section type.
			 *
			 * @access public
			 * @var string
			 */
			public $type = 'kirki_installer';

			/**
			 * The plugin install URL.
			 *
			 * @access private
			 * @var string
			 */
			public $plugin_install_url;

			/**
			 * Render the section.
			 *
			 * @access protected
			 */
			protected function render() {

				// Don't proceed any further if the user has dismissed this.
				if ( $this->is_dismissed() ) {
					return;
				}

				// Determine if the plugin is not installed, or just inactive.
				$plugins   = get_plugins();
				$installed = false;
				foreach ( $plugins as $plugin ) {
					if ( 'Kirki' === $plugin['Name'] || 'Kirki Toolkit' === $plugin['Name'] ) {
						$installed = true;
					}
				}
				$plugin_install_url = $this->get_plugin_install_url();
				$classes            = 'cannot-expand accordion-section control-section control-section-themes control-section-' . $this->type;
				?>
                <li id="accordion-section-<?php echo esc_attr( $this->id ); ?>"
                    class="<?php echo esc_attr( $classes ); ?>"
                    style="border-top:none;border-bottom:1px solid #ddd;padding:7px 14px 21px 14px;text-align:left;">
					<?php if ( ! $installed ) : ?>
						<?php $this->install_button(); ?>
					<?php else : ?>
						<?php $this->activate_button(); ?>
					<?php endif; ?>
					<?php $this->dismiss_button(); ?>
                </li>
				<?php
			}

			/**
			 * Check if the user has chosen to hide this.
			 *
			 * @static
			 * @access public
			 * @since  1.0.0
			 * @return bool
			 */
			public static function is_dismissed() {
				// Get the user-meta.
				$user_id   = get_current_user_id();
				$user_meta = get_user_meta( $user_id, 'dismiss-kirki-recommendation', true );

				return ( true === $user_meta || '1' === $user_meta || 1 === $user_meta );
			}

			/**
			 * Adds the install button.
			 *
			 * @since 1.0.0
			 * @return void
			 */
			protected function install_button() {
				?>
                <p style="text-align:left;margin-top:0;">
					<?php esc_attr_e( 'Please install the Kirki plugin to take full advantage of this theme\'s customizer capabilities', 'startup-pro' ); ?>
                </p>
                <a class="install-now button-primary button" data-slug="kirki"
                   href="<?php echo esc_url_raw( $this->get_plugin_install_url() ); ?>"
                   aria-label="<?php esc_attr_e( 'Install Kirki Toolkit now', 'startup-pro' ); ?>"
                   data-name="Kirki Toolkit">
					<?php esc_html_e( 'Install Now', 'startup-pro' ); ?>
                </a>
				<?php
			}

			/**
			 * Adds the install button.
			 *
			 * @since 1.0.0
			 * @return void
			 */
			protected function activate_button() {
				?>
                <p style="text-align:left;margin-top:0;">
					<?php esc_attr_e( 'You have installed Kirki. Activate it to take advantage of this theme\'s features in the customizer.', 'startup-pro' ); ?>
                </p>
                <a class="activate-now button-primary button" data-slug="kirki"
                   href="<?php echo esc_url_raw( self_admin_url( 'plugins.php' ) ); ?>"
                   aria-label="<?php esc_attr_e( 'Activate Kirki Toolkit now', 'startup-pro' ); ?>"
                   data-name="Kirki Toolkit">
					<?php esc_html_e( 'Activate Now', 'startup-pro' ); ?>
                </a>
				<?php
			}

			/**
			 * Adds the dismiss button and script.
			 *
			 * @since 1.0.0
			 * @return void
			 */
			protected function dismiss_button() {

				// Create the nonce.
				$ajax_nonce = wp_create_nonce( 'dismiss-kirki-recommendation' );

				// Show confirmation dialog on dismiss?
				$show_confirm = true;
				?>
                <a class="kirki-installer-dismiss button-secondary button"
                   data-slug="kirki" href="#"
                   aria-label="<?php esc_attr_e( 'Don\'t show this again', 'startup-pro' ); ?>"
                   data-name="<?php esc_attr_e( 'Dismiss', 'startup-pro' ); ?>">
					<?php esc_attr_e( 'Don\'t show this again', 'startup-pro' ); ?>
                </a>

                <script type="text/javascript">
                    jQuery(document).ready(function () {
                        jQuery('.kirki-installer-dismiss').on('click', function (event) {

                            event.preventDefault();

							<?php if ( $show_confirm ) : ?>
                            if (!confirm('<?php esc_attr_e( 'Are you sure? Dismissing this message will hide the installation recommendation and you will have to manually install and activate the plugin in the future.', 'startup-pro' ); ?>')) {
                                return;
                            }
							<?php endif; ?>

                            jQuery.post(ajaxurl, {
                                action: 'kirki_installer_dismiss',
                                security: '<?php echo esc_attr( $ajax_nonce ); ?>',
                            }, function (response) {
                                jQuery('#accordion-section-kirki_installer').remove();
                            });
                        });
                    });
                </script>
				<?php
			}

			/**
			 * Get the plugin install URL.
			 *
			 * @access private
			 * @return string
			 */
			private function get_plugin_install_url() {
				if ( ! $this->plugin_install_url ) {
					// Get the plugin-installation URL.
					$this->plugin_install_url = add_query_arg(
						array(
							'action' => 'install-plugin',
							'plugin' => 'kirki',
						),
						self_admin_url( 'update.php' )
					);
					$this->plugin_install_url = wp_nonce_url( $this->plugin_install_url, 'install-plugin_kirki' );
				}

				return $this->plugin_install_url;
			}
		}
	}

	// Early exit if the user has dismissed the notice.
	if ( is_callable( array(
			'Kirki_Installer_Section',
			'is_dismissed'
		) ) && Kirki_Installer_Section::is_dismissed() ) {
		return;
	}

	$wp_customize->add_section(
		new Kirki_Installer_Section(
			$wp_customize, 'kirki_installer', array(
				'title'      => '',
				'capability' => 'install_plugins',
				'priority'   => 0,
			)
		)
	);
	$wp_customize->add_setting(
		'kirki_installer_setting', array(
			'sanitize_callback' => '__return_true',
		)
	);
	$wp_customize->add_control(
		'kirki_installer_control', array(
			'section'  => 'kirki_installer',
			'settings' => 'kirki_installer_setting',
		)
	);

}

add_action( 'wp_ajax_kirki_installer_dismiss', 'startup_kirki_installer_dismiss' );
/**
 * Handles dismissing the plugin-install/activate recommendation.
 * If the user clicks the "Don't show this again" button, save as user-meta.
 *
 * @since 1.1.0
 *
 * @return void
 */
function startup_kirki_installer_dismiss() {
	check_ajax_referer( 'dismiss-kirki-recommendation', 'security' );
	$user_id = get_current_user_id();

	if ( update_user_meta( $user_id, 'dismiss-kirki-recommendation', true ) ) {
		echo 'success! :-)';
		wp_die();
	}

	echo 'failed :-(';
	wp_die();
}
